### ISSUE TITLE
Get the diagnostic context in the event. 

### ISSUE THREAD

**Issue by tdeverdiere** [netmikey/logunit#5]
That would be cool to have the map of diagnostic context in the final log event. This is equivalent to ThreadContext Map in log4j2, and to MDC in logback.

That would allow to test that `ThreadContext.put()` and `MDC.put` is being called.

**Comment by netmikey** [comment]
LogUnit uses Slf4j's `LoggingEvent` interface to capture logging events because we'd rather not invent yet another LoggingEvent interface. Slf4j's interface doesn't provide a way to access the MDC unfortunately. As far as I can see, the new [Slf4j v2 `LoggingEvent`](https://github.com/qos-ch/slf4j/blob/master/slf4j-api/src/main/java/org/slf4j/event/LoggingEvent.java) doesn't provide it either.

The underlying Log4j2 and Logback events both provide access to their MDC implementation though. You might want to check with the Slf4j project to see if they'd be willing to add a `getMDC()` method of some sort to their `LoggingEvent` interface.

**Comment by tdeverdiere** [comment]
Yes you are right. Or maybe we could have a wrapper in logunit ? 

I created an issue on Slf4j : https://github.com/qos-ch/slf4j/discussions/345

**Comment by netmikey** [comment]
We could also inherit / extend slf4j's `LoggingEvent` interface but let's wait for the slf4j project's opinion first.
